### DISCUSSION TITLE
Extracting Twiss parameters

### DISCUSSION THREAD

**Discussion by dmargarone** [ocelot-collab/ocelot#229] [Q&A] [answered]
Hi, 

I described a particle array as follows:

```
p_array_init = generate_parray(sigma_x=1.0e-3, sigma_tau=0.5e-3,
                               sigma_px=7e-5, energy=0.1, nparticles=20000, charge=1e-09)
```

How can I get the Twiss parameter of the electron beam corresponding to this "p_array_init".

Thank you.
Daniele

**Comment by st-walker** [comment, marked as answer]
`get_envelope`

**Comment by dmargarone** [comment]
@st-walker , 

Thank you. 

Daniele

**Comment by dmargarone** [comment]
Hi @st-walker , I have one confusion about the emittance of the electron beam that I get from the "get_envelope" and what I saw from "show_e_beam".

```
p_array_init = generate_parray(sigma_x=100.0e-6, sigma_y=100.0e-6, sigma_tau=100.0e-6,
                               sigma_px=7e-5, chirp=0.0002, energy=Eb,
                               nparticles=20000, charge=1e-09)


# get twiss parameters of the electron beam
Twiss = get_envelope(p_array_init)

print("")
print("Twiss parameters corresponding to p_array_init")
print("")
print(Twiss)

show_e_beam(p_array_init, figsize=(9,6))
```

show_e_beam gives emit_x = 1.36 mm. mrad and emit_y = 1.37 mm. mrad, where as Twiss parameters from get_envelope gives emit_x = 6.949964803224176e-09 and emit_y = 6.982866393378305e-09. 

Why there is so big difference in show_e_beam and get_envelope values.

**Reply by st-walker** [reply]
can u show plot?

**Comment by dmargarone** [comment]
@st-walker , yes here it is attached.

<img width="1239" alt="ebeam_distribution" src="https://github.com/ocelot-collab/ocelot/assets/88077574/a5f593b4-53e9-4262-b689-894fb7211192">

**Reply by st-walker** [reply]
perhaps you are comparing two different quantities: geometric emittance and normalised emittance.  in output of `get_envelope` there is `emit_x` and `emit_xn`.  I think the plot will probably be `emit_xn` and `emit_yn`, which is a factor of relativistic gamma * beta greater than geometric emittance.

**Comment by dmargarone** [comment]
And the electron beam energy in "p_array_init" is  Eb = 0.1
